module/persistent.lua

```diff
@@ -0,0 +1,186 @@
+local awful = require("awful")
+local gears = require("gears")
+
+local capi =
+{
+    awesome = awesome,
+    root = root,
+    screen = screen,
+    client = client
+}
+local ipairs = ipairs
+local table = table
+local tonumber = tonumber
+local tostring = tostring
+
+local persistent = { mt = {} }
+local instance = nil
+
+local function get_xproperty(name, type)
+    capi.awesome.register_xproperty(name, type)
+    return capi.awesome.get_xproperty(name)
+end
+
+local function set_xproperty(name, type, value)
+    capi.awesome.register_xproperty(name, type)
+    capi.awesome.set_xproperty(name, value)
+end
+
+local function client_get_xproperty(client, name, type)
+    local xprop = "bling.client." .. name
+    capi.awesome.register_xproperty(xprop, type)
+    -- print(xprop .. "  " .. tostring(client:get_xproperty(xprop)))
+    return client:get_xproperty(xprop)
+end
+
+local function client_set_xproperty(client, name, type, value)
+    local xprop = "bling.client." .. name
+    capi.awesome.register_xproperty(xprop, type)
+    client:set_xproperty(xprop, value)
+end
+
+local function save()
+    set_xproperty("tag_index", "number", awful.screen.focused().selected_tag.index)
+
+    for index, entry in ipairs(capi.root.tags()) do
+        local master_width_property = "tag_" .. entry.index .. "_master_width_factor"
+        set_xproperty(master_width_property, "string", tostring(entry.master_width_factor))
+
+        local layout_property = "tag_" .. entry.index .. "_layout"
+        set_xproperty(layout_property, "number", awful.layout.get_tag_layout_index(entry))
+    end
+
+    for index, client in ipairs(capi.client.get()) do
+        client_set_xproperty(client, "screen", "number", client.screen.index)
+        client_set_xproperty(client, "hidden", "boolean", client.hidden)
+        client_set_xproperty(client, "minimized", "boolean", client.minimized)
+        client_set_xproperty(client, "above", "boolean", client.above)
+        client_set_xproperty(client, "ontop", "boolean", client.ontop)
+        client_set_xproperty(client, "below", "boolean", client.below)
+        client_set_xproperty(client, "fullscreen", "boolean", client.fullscreen)
+        client_set_xproperty(client, "maximized", "boolean", client.maximized)
```

Comment:
Isn't stuff like maximized and minimized persistency already taken care of by awesome?

Reply:
> Isn't stuff like maximized and minimized persistency already taken care of by awesome?

It is, and so is floating I think, but if you have any rules that overrides those, whatever you had set there will override the  values from the previous session

Reply:
Ah alright makes sense :thumbsup: